gcp/src/main/java/org/apache/iceberg/gcp/biglake/BigLakeCatalog.java

```diff
@@ -0,0 +1,398 @@
+    properties.forEach(optionsBuilder::removeParameters);
+    client.updateDatabaseParameters(getDatabaseName(namespace), optionsBuilder.getParametersMap());
+    return true;
+  }
+
+  @Override
+  public Map<String, String> loadNamespaceMetadata(Namespace namespace) {
+    if (namespace.levels().length == 0) {
+      // Calls getCatalog to check existence. BLMS catalog has no metadata today.
+      client.getCatalog(catalogName);
+      return new HashMap<String, String>();
+    } else if (namespace.levels().length == 1) {
+      return getMetadata(getDatabase(namespace));
+    } else {
+      throw new IllegalArgumentException(invalidNamespaceMessage(namespace));
+    }
+  }
+
+  @Override
+  public String name() {
+    return catalogPulginName;
+  }
+
+  @Override
+  protected Map<String, String> properties() {
+    return catalogProperties == null ? ImmutableMap.of() : catalogProperties;
+  }
+
+  @Override
+  public void setConf(Object conf) {
+    this.conf = conf;
+  }
+
+  private String getDatabaseLocation(String dbId) {
+    String warehouseLocation =
+        LocationUtil.stripTrailingSlash(
+            catalogProperties.get(CatalogProperties.WAREHOUSE_LOCATION));
+    Preconditions.checkNotNull(warehouseLocation, "Data warehouse location is not set");
+    return String.format("%s/%s.db", LocationUtil.stripTrailingSlash(warehouseLocation), dbId);
+  }
+
+  private static TableIdentifier getTableIdentifier(Table table) {
+    TableName tableName = TableName.parse(table.getName());
+    return TableIdentifier.of(Namespace.of(tableName.getDatabase()), tableName.getTable());
+  }
+
+  private static Namespace getNamespace(Database db) {
+    return Namespace.of(DatabaseName.parse(db.getName()).getDatabase());
+  }
+
+  private TableName getTableName(String dbId, String tableId) {
+    return TableName.of(projectId, location, catalogId, dbId, tableId);
+  }
+
+  private String getDatabaseId(Namespace namespace) {
+    Preconditions.checkArgument(
+        namespace.levels().length == 1,
+        "BigLake database namespace must use format <catalog>.<database>, invalid namespace: %s",
+        namespace);
+    return namespace.level(0);
+  }
+
+  private DatabaseName getDatabaseName(Namespace namespace) {
+    return DatabaseName.of(projectId, location, catalogId, getDatabaseId(namespace));
+  }
+
+  private Database getDatabase(Namespace namespace) {
+    return client.getDatabase(getDatabaseName(namespace));
+  }
+
+  private static Map<String, String> getMetadata(Database db) {
+    HiveDatabaseOptions options = db.getHiveOptions();
+    Map<String, String> result = Maps.newHashMap();
+    result.putAll(options.getParameters());
+    result.put("location", options.getLocationUri());
+    return result;
+  }
+
+  private static String invalidNamespaceMessage(Namespace namespace) {
+    return String.format(
+        "BigLake catalog namespace can have zero (catalog) or one level (database), invalid"
+            + " namespace: %s",
```

Comment:
Can you make this shorter?

Also, this is only used in places that have already checked for a 0-length namespace. I think this only needs to worry about namespaces that are too long.
```java
return String.format("Invalid namespace (too long): %s", namespace);
```

Reply:
Sure, done.